internal/pkg/config/config.go

```diff
@@ -108,6 +108,20 @@ func (c Config) IsSource(t types.Type) bool {
 	return false
 }
 
+func (c Config) IsSourceField(typ types.Type, fld *types.Var) bool {
+	n, ok := typ.(*types.Named)
+	if !ok {
+		return false
+	}
+
+	for _, p := range c.Sources {
```

Comment:
`IsSourceField` should take a `Source`, right? Otherwise, we could match identically named fields to different Sources. In other words, if field-x is PII in one type it may not be PII in another type.

I realize that this is an existing bug, bug since we are touching this area now, do you think we should fix his first?

Reply:
`IsSourceField` is mirroring `IsSourceFieldAddr` -- within the loop `p.match(n)` matches package and type, and `p.FieldRE.MatchString` matches the field name.  It's conditional on satisfying both for the same `p` in the configuration's source matchers.  The only risk of cross-identification is if the config's regexp are wider than intended, which this analysis could verify / validate.